src/components/TextInput.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { FormEventHandler } from 'react';
+import styled from 'styled-components';
+
+import { rem } from '../constants';
+import colors from '../styles/colors';
+
+interface TextInputProps {
+	id: string;
+	disabled?: boolean;
+	label?: string;
+	name?: string;
+	placeholder?: string;
+	size?: 'medium' | 'small';
+	value: string;
+	onInput: FormEventHandler;
+}
+
+export default function TextInput({
+	id,
+	disabled = false,
+	label,
+	onInput,
+	placeholder,
+	size = 'medium',
+	value,
+	name,
+}: TextInputProps) {
+	return (
+		<StyledInputWrapper>
+			<StyledTextInput
+				id={id}
+				disabled={disabled}
+				onInput={onInput}
+				placeholder={placeholder}
+				inputSize={size}
+				value={value}
+				name={name}
+			/>
+			{label && <StyledLabel htmlFor={id}>{label}</StyledLabel>}
+		</StyledInputWrapper>
+	);
+}
+
+const StyledInputWrapper = styled.div`
+	position: relative;
+	display: flex;
+	flex-direction: column;
+`;
+
+const StyledLabel = styled.label`
+	order: 1;
+	display: inline-block;
+	padding-bottom: 6px;
+	line-height: 1;
+	font-size: ${rem}px;
+	color: ${colors.grey};
+	cursor: pointer;
+`;
+
+const sizes = {
+	medium: {
+		padding: '13px 23px 13.5px',
```

Comment:
just go with 13px 23px. Figma sometimes got some weird calculation going on. We don't need to do everything as in figma to 100%